devolutions-gateway/src/api/network_scan/ipconfig.rs

```diff
@@ -0,0 +1,75 @@
+use std::net::IpAddr;
+
+use axum::body::Body;
+use axum::response::Response;
+use network_scanner::interfaces::{MacAddr, NetworkInterface};
+use serde::Serialize;
+
+use crate::http::HttpError;
+
+pub async fn handler(_netscan_claim: crate::extract::NetScanToken) -> Result<Response, HttpError> {
+    let res = network_scanner::interfaces::get_network_interfaces().map_err(|e| {
+        tracing::error!("failed to get network interfaces: {:?}", e);
+        HttpError::internal().build(e)
+    })?;
```

Comment:
suggestion: This is how it’s supposed to be used. The error will be logged automatically.
```suggestion
    let res = network_scanner::interfaces::get_network_interfaces()
        .map_err(HttpError::internal().msg("failed to get network interfaces"))?;
```